types/breejs__later/index.d.ts

```diff
@@ -0,0 +1,683 @@
+         */
+        second: TimePeriod;
+        s: TimePeriod;
+        /**
+         * Minute time period information provider.
+         */
+        minute: TimePeriod;
+        m: TimePeriod;
+        /**
+         * Hour time period information provider.
+         */
+        hour: TimePeriod;
+        h: TimePeriod;
+        /**
+         * Day time period information provider.
+         */
+        day: TimePeriod;
+        D: TimePeriod;
+        /**
+         * Day of week time period information provider.
+         */
+        dayOfWeek: TimePeriod;
+        dw: TimePeriod;
+        d: TimePeriod;
+        /**
+         * Day of week in month time period information provider.
+         */
+        dayOfWeekCount: TimePeriod;
+        dc: TimePeriod;
+        /**
+         * Day in year time period information provider.
+         */
+        dayOfYear: TimePeriod;
+        dy: TimePeriod;
+        /**
+         * Week of month time period information provider.
+         */
+        weekOfMonth: TimePeriod;
+        wm: TimePeriod;
+        /**
+         * Week of year from ISO 8601 time period information provider.
+         */
+        weekOfYear: TimePeriod;
+        wy: TimePeriod;
+        /**
+         * Month time period information provider.
+         */
+        month: TimePeriod;
+        M: TimePeriod;
+        /**
+         * Year time period information provider.
+         */
+        year: TimePeriod;
+        Y: TimePeriod;
+        /**
+         * Full date time period information provider.
+         */
+        fullDate: TimePeriod;
+        fd: TimePeriod;
+
+        /**
+         * Constant for the number of milliseconds in a second.
+         */
+        SEC: 1000;
+        /**
+         * Constant for the number of milliseconds in a minute.
+         */
+        MIN: 60_000;
+        /**
+         * Constant for the number of milliseconds in an hour.
+         */
+        HOUR: 3_600_000;
+        /**
+         * Constant for the number of milliseconds in a day.
+         */
+        DAY: 86_400_000;
+        /**
+         * Constant for the number of milliseconds in a week.
+         */
+        WEEK: 604_800_000;
+        /**
+         * Constant for the number of days in each month.
+         */
+        DAYS_IN_MONTH: [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
+        /**
+         * Constant for never.
+         */
+        NEVER: 0;
+
+        /**
+         * Later Modifiers:
+         *
+         * This type can be easily extended to include any custom IModifiers that you desire.
+         * These can then be used to create schedules of your own custom type.
+         *
+         * interface IGandalfsLaterModifier extends Later.IModifierStatic {
+         *     duringTheThirdAge: IModifier
+         * }
+         *
+         * Be sure to use this interface when dealing with Later.modifier
+         */
+        modifier: ModifierStatic;
+    }
+}
+
+declare const later: Later.Static;
```

Comment:
I took another look at the sources. Since `later` is essentially just a namespace, you could drop `Later.Static` and make all of the members of `Later.Static` just exports of the `Later` namespace, i.e.:

```ts
declare namespace Later {
    ... // everything currently in the Later namespace
    
    /**
     * Array methods
     */
    namespace array {
      ... // everything that was in `ArrayStatic`, except as a namespace
    }

    /**
     * Generates instances from schedule data.
     */
    function schedule(input: any): Schedule;

    /**
     * @param schedule The schedule to be parsed
     */
    function compile(schedule: Schedule): TimePeriod;

    /**
     * Parse
     * For generating schedule data.
     */
    namespace parse {
      ... // everything that was in `ParseStatic`, except as a namespace
    }

    ... // everything else that was in `Static`, except as a namspace
}

export = Later; // export the namespace itself.
```

Reply:
Hi @rbuckton! 👋 

I nudged the Renovate team into trying to upstream the `@breejs/later` types again:

- https://github.com/renovatebot/renovate/issues/18085

> I took another look at the sources. Since later is essentially just a namespace, you could drop Later.Static and make all of the members of Later.Static just exports of the Later namespace, i.e.:

Are those changes _needed_ before this PR gets merged? I see your stale bot is going to close the PR soon.

I propose:

1. Merge what we have now
1. Create a new issue on the Definitely Typed project, with your suggested improvements
1. Assign @JamieMagee to the new issue

Note that @JamieMagee already tried to get the `breejs` types into the project before:

- https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56048

Greetings,

HonkingGoose

Reply:
@HonkingGoose I think I can address it in this PR, it just dropped off my radar. Thanks for the reminder!

@rbuckton I think I've made the changes you suggested?